File: sdks/java/src/main/java/com/digows/whatsappgateway/session/SessionDesiredState.java
package com.digows.whatsappgateway.session;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum SessionDesiredState
{
  ACTIVE("active"),
  PAUSED("paused"),
  STOPPED("stopped");

  private final String wireValue;

  SessionDesiredState(String wireValue)
  {
    this.wireValue = wireValue;
  }

  @JsonValue
  public String getWireValue()
  {
    return wireValue;
  }

  @JsonCreator
  public static SessionDesiredState fromWireValue(String wireValue)
  {
    for (SessionDesiredState value : values())
    {
      if (value.wireValue.equals(wireValue))
      {
        return value;
      }
    }

    throw new IllegalArgumentException("Unsupported session desired state \"" + wireValue + "\".");
  }
}

File: runtime/src/domain/entities/session/SessionDesiredState.ts
/**
 * Desired operational policy for a mirrored WhatsApp session.
 * The embedded control plane reconciles the runtime towards this state.
 */
export enum SessionDesiredState {
  Active = 'active',
  Paused = 'paused',
  Stopped = 'stopped',
}

export function parseSessionDesiredState(value: string): SessionDesiredState {
  switch (value) {
    case SessionDesiredState.Active:
      return SessionDesiredState.Active;
    case SessionDesiredState.Paused:
      return SessionDesiredState.Paused;
    case SessionDesiredState.Stopped:
      return SessionDesiredState.Stopped;
    default:
      throw new Error(`Unsupported session desired state "${value}".`);
  }
}
